src/compile/legend/rules.ts

```diff
@@ -26,12 +27,11 @@ export function values(legend: Legend) {
   return vals;
 }
 
-export function type(legend: Legend, fieldDef: FieldDef, channel: Channel) {
+export function type(legend: Legend, timeUnit: TimeUnit, channel: Channel, scaleType: ScaleType) {
   if (legend.type) {
     return legend.type;
   }
-
-  if (channel === COLOR && !fieldDef.bin && !fieldDef.timeUnit && contains([QUANTITATIVE, TEMPORAL], fieldDef.type)) {
+  if (channel === COLOR && (!timeUnit || containsTimeUnit(timeUnit, 'year')) && contains<ScaleType>(['time', 'utc', 'sequential'], scaleType)) {
```

Comment:
I meant replace  `containsTimeUnit(timeUnit, 'year')` with scale type check just for temporal case.  

I think the new logic seems wrong.   You can sensibly use linear, log, pow with quantitative field too.  I think the old `contains([QUANTITATIVE, TEMPORAL], fieldDef.type)` was sensible.